alf/config_util.py

```diff
@@ -306,6 +344,11 @@ def config1(config_name, value, mutable=True, raise_if_used=True):
             immutable value to an existing immutable value.
         raise_if_used (bool): If True, ValueError will be raised if trying to
             config a value which has already been used.
+        sole_init (bool): If True, the config value can only be set once. Any
+            previous or future calls will raise a ValueError. This is helpful
+            in enforcing a singular point of initialization, thus eliminating
+            any potential side effects from possible prior or future overrides.
+            This flag overrides the mutable flag.
```

Comment:
This flag only overwrites the mutable flag if it's True?

Reply:
Right. Good catch. I've clarified this.